.claude/skills/webapp-testing/scripts/with_server.py

```
@@ -0,0 +1,106 @@
+    for cmd, port in zip(args.servers, args.ports):
+        servers.append({'cmd': cmd, 'port': port})
+
+    server_processes = []
+
+    try:
+        # Start all servers
+        for i, server in enumerate(servers):
+            print(f"Starting server {i+1}/{len(servers)}: {server['cmd']}")
+
+            # Use shell=True to support commands with cd and &&
+            process = subprocess.Popen(
```

Ruff (0.15.0)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)